### DISCUSSION TITLE
How does onTouch work?

### DISCUSSION THREAD

**Discussion by CubitsDev** [Minestom/Minestom#676] [Help]
Hi there,

I'm attempting to use the `onTouch` method in a BlockHandler. However, when touching the block, it appears the method isn't being run.

Handler:
```
public class BarrierHandler implements BlockHandler {

    @Override
    public void onTouch(@NotNull Touch touch) {
        Player player = (Player) touch.getTouching();
        player.teleport(HubServer.getSpawnPos());
    }

    @Override
    public @NotNull NamespaceID getNamespaceId() {
        return NamespaceID.from("hub:barrier");
    }
}
```

Register:
`MinecraftServer.getBlockManager().registerHandler("minecraft:barrier", BarrierHandler::new);`

Any ideas? Is this **not** how onTouch functions?

**Comment by EpicPlayerA10** [comment]
`onTouch` fires when player is in block. It is useful when we want to check if player touches e.g fire.

**Comment by gabber235** [comment]
In the previous #help in the discord, I also asked this question. `onTouch` should be seen more as an `onIntersect`. It is fired when an entity intersects with the block, eg is in the block. It can be used with non-solid blocks like air, buttons, fire, or blocks that don't have a full hitbox like fences. 
Note that **this check is not run every tick** for performance reasons, thus it might not trigger when an entity goes through the block too fast. 

Your example with a barrier will not work, since there is no way for a player to be inside a barrier. You would have to find a workaround for your particular use case. A possible solution could be to add the handler on air blocks on the sides of the barrier.
